### ISSUE TITLE
mandatory delays for horizontal Review

### ISSUE THREAD

**Issue by frivoal** [w3c/charter-drafts#219]
"Section 3: Coordination" of the [previous csswg charter](https://www.w3.org/Style/2016/css-2016) said this:

> Invitation for review must be issued during each major standards-track document transition, including FPWD and CR, and should be issued when major changes occur in a specification.

The [proposed 2019 charter](https://w3c.github.io/charter-drafts/css-2019.html) changes this to:

> Invitation for review must be issued during each major standards-track document transition, including FPWD and at least 3 months before CR, and should be issued when major changes occur in a specification.

I support saying that the review must be done **before** CR, rather than during CR, and agree that horizontal review groups should be given enough time to do their job properly. However, I think we should be cautious constraining what the chairs are allowed to do with hard requirements on review periods that go beyond what the process says.

Somewhat artificial example: let's invitation for review has been issue for a spec that intends to go to CR, and review has been received after 2 months. That particular spec is urgent, as we want to publish before some external deadline. Having received review, we'd like to publish now, but since the charter says *must* on 3 months, we cannot. Furthermore, 3 months takes into some w3c team moratorium (maybe TPAC, or the holidays season), which would turn the 3 months into 3 and a half. Even though everything, including review, has been done, the charter forces us to wait 1.5 extra months. This isn't great.

I propose replacing by this:

> Invitation for review must be issued in connection with each major standards-track document transition, including when publishing FPWD and before transitioning to CR (at least 3 months before is recommended), and should be issued when major changes occur in a specification.

**Comment by astearns** [comment]
I worry that loosening the restriction there would encourage people to send out invitations for review too late for feedback before CR (we sent it out the week before, didn't hear back, so we can address the feedback in an updated CR).

What about this to satisfy your example?

> Invitation for review must be issued in connection with each major standards-track document transition, including when publishing FPWD and at least 3 months before CR (this interval can be waived when sufficient feedback is received). Invitation for review should also be issued when major changes occur in a specification.

**Comment by frivoal** [comment]
I agree it's an improvement over the original, which deals with the example I gave.

However, I'd rather leave this at the discretion of the chairs to appraise each situation individually, rather than hard-code the cases when exceptions can be granted. "is recommended" (same as RFC2119 SHOULD) in my proposed wording still sets the expectation, but gives the WG/chairs flexibility in granting exceptions.

**Comment by r12a** [comment]
This advice isn't saying that you should complete the spec, then wait for 3 months for wide review.  That would be the equivalent of reintroducing the Last Call period.

What it's saying is that, if you have a date for transition to CR ensure that you *ask for* horizontal review at least x amount of time before that date.  

If you've done self-reviews, checked things prior to FPWD, and checked with HR groups when major changes enter the spec, then with luck there won't be a huge effort involved for that review, and it will probably not take 3 months (or the 60 days mentioned in recent discussions). On the other hand, the WG may ask the HR group to do another final review fairly close to CR to check any changes that took place over the time since the previous review (without extending the CR deadline).

The main objective of this 3 months before CR is to specifically avoid falling back into the LC scenario.  Here's something i wrote in another thread recently:

> I looked back over the review requests i18n has had in the past 6 months (which is representative of previous years) and i see that **all but one requests were sent two, or with luck three, weeks before a fixed date for transition to CR**.  _The LC process wasn't good, but this is worse!_

**Comment by frivoal** [comment]
I agree with the intent, I am a little skeptical of tying the chairs hands and preventing them from granting well reasoned exceptions.

**Comment by r12a** [comment]
So how about something like the following, which tries to phrase it in a way that indicates that this is not an unmoveable 3-month window:

> Invitation for review must be issued in connection with each major standards-track document transition, including when publishing FPWD and at least 3 months before CR (the date of transition to CR can be always be brought closer as long as there has been sufficient horizontal review). Invitation for review should also be issued when major changes occur in a specification.

**Comment by fantasai** [comment]
@r12a Your parenthetical contradicts, rather than adjusts, the phrase “at least 3 months before”, so that's not going to work.

Also, the current wording could be satisfied by just having an FPWD review request, because that's usually at least 3 months before CR. :) So altogether, it's not very good wording.

I would suggest something like

> Invitation for review must be issued in conjunction with FPWD, should be issued when major changes occur in a specification, and must be issued with sufficient time (which should be at least three months, unless the horizontal review group has indicated otherwise) before a CR transition date that a review can be completed on substantially the same document that will become the CR.

**Comment by plehegar** [comment]
I liked the latest proposed change. See #233 .

**Comment by r12a** [comment]
I don't like the following aspects of Fantasai's version:

1. the text implies that the horizontal review groups (which should be plural) need to be consulted each time a spec is approaching CR to determine the length of time needed for this review period.  That is not something we, and i suspect others, want to do.  And anyway, the HR group may not be able to determine how much time is needed before CR until they do the review.
2. requiring review on substantially the same document that will become the CR implies that the 3 month period is a LC phase, with all the old problems that that approach entailed.  The new idea for reviewing is to work with the WG and catch issues earlier, and then through ongoing dialogue check that issues have been resolved before the transition.

The way i see it is simple. The WG generally has an idea of a date when it wants to go to CR.  Before that date they need to issue a request for review to the horizontal groups which allows time for the HR group to schedule a review and for any necessary changes in direction to be spotted and acted upon by the time that date is reached.  

This is just about ensuring that we have sufficient lead time to make that happen. I don't think the HR groups require the document to be stable, although for most specs it's likely to be close to final anyway by that time. 

Florian's concern as i understood it was that a group may want to bring forward the CR date, and i think that should be possible as long as any issues raised during horizontal review have been addressed.

**Comment by r12a** [comment]
We just discussed this thread in the W3C Strategy call, and the other folks involved in HR groups on that call also shared the same concern to move away from the old Quality Control model (ie. develop then inspect) to a Quality Assurance model (ie. work with the WG to prevent issues arising). They also agreed that the spec doesn't need to be essentially the same as that which goes to CR when the 3 month period starts.

It occurred to me during that call that perhaps the word 'review' carries too much unhelpful baggage. I'm not sure how one would replace that word with something meaningful in a useful way, but it does suggest the 'you finish this then we'll take a look' approach.  Whereas what we have in mind is more that, from 3 months prior to CR transition, HR groups will  **collaborate with the WG** to ensure that horizontal concerns are being addressed.

Of course, part of that will usually involve reviewing the currently available spec, but (as is already the case) the review itself is usually only a brief activity and the real goal is the interaction between the groups around asking questions, discussing approaches, explaining issues, and agreeing any needed changes. The review itself is just a handy way for the HR group to bring themselves up to speed.

And by the way, if the WG has done a self-review at FPWD, bears in mind horizontal stuff arising from that, and shares with the HR group any significant deviations, then the work to be done in these 3 months hopefully becomes much less of a shock to the system and more a tweaking of remaining details.

**Bottom line:** we're trying to get away from the old model of WGs developing their spec then handing over to HR for inspection during a fixed, add-on time period.  We're trying to get to early identification of HR issues, and to HR groups working alongside the WG in order to tidy up the remaining issues. 3 months out from the planned CR date seems to be a good time to ensure that that collaboration has kicked in, but there is no fixed period involved after that collaboration starts other than the time needed to address any HR issues, if there are any.  If they are addressed quickly, the CR transition can be brought forward.

**Comment by plehegar** [comment]
Alright, my current thinking is to stick to the current wording in the proposed charter, without closing this issue. If new wording emerges before the end of the AC review, we can always update the proposed charter imho.

**Comment by frivoal** [comment]
> the text implies that the horizontal review groups (which should be plural) need to be consulted each time a spec is approaching CR to determine the length of time needed for this review period.

It does not. It says that they should be consulted if we want to make the period shorter than 3 months. I imagine the way this would typically go is that a WG asks them for review, and if they come back with a completed review after 2 months, they'd also indicate that the review is done and there's no need to wait for one more.

> requiring review on substantially the same document that will become the CR implies that the 3 month period is a LC phase, with all the old problems that that approach entailed. The new idea for reviewing is to work with the WG and catch issues earlier, and then through ongoing dialogue check that issues have been resolved before the transition.

So, if we call for review on the document, get the review done, then spend the next 4 years changing most things about the document, we're good to go because we did ask more than 3 month before CR? Besides, since all groups are required by the charter to ask for review at FPWD, then no matter how much the document changes after that, they've fullfilled the requirement of asking for review 3 months in advance as long as there's more than 3 months between FPWD and CR... This is rather pointless.


> The WG generally has an idea of a date when it wants to go to CR.

Not really.  CR is rarely date driven. The WG wants to go with CR when it's done dealing with issues that block it from doing so. Various people may have a guess of how long that's going to take, but their guesses aren't typically all that accurate.

> Florian's concern as i understood it was that a group may want to bring forward the CR date, and **i think that should be possible** as long as any issues raised during horizontal review have been addressed.

Right, and that is generally allowed by the process, but explicitly forbidden in by this bit of charter text I'm opposing.

> from 3 months prior to CR transition, HR groups will collaborate with the WG to ensure that horizontal concerns are being addressed.

That sounds good. But the text currenlty proposed in the charter does not do that. Therefore: 
> my current thinking is to stick to the current wording in the proposed charter, without closing this issue. If new wording emerges before the end of the AC review, we can always update the proposed charter imho.

That sounds like the wrong thing to do. What r12a wants is not what is in the charter. I (and fantasai) don't like what's in the charter. r12a doesn't like what fantasai proposed, but he's not defending the charter either. Until we can agree on something (and I am hopeful that we can, since r12a's points look reasonable to me), we should not place on ourselves constraints beyond what the Process asks. I suggest deleting that text until new wording emerges.

**Comment by r12a** [comment]
> So, if we call for review on the document, get the review done, then spend the next 4 years changing most things about the document, we're good to go because we did ask more than 3 month before CR? Besides, since all groups are required by the charter to ask for review at FPWD, then no matter how much the document changes after that, they've fullfilled the requirement of asking for review 3 months in advance as long as there's more than 3 months between FPWD and CR... This is rather pointless.

An easy way to fix that is to remove 'at least' from 'at least 3 months' , and it could be made clearer by saying 3 months before the expected CR transition date – something i was inclined to suggest anyway, actually.  That would probably simply things significantly, help change the perception of what is going on here, and allow the invitation to be triggered automatically by tools. 

Bear in mind that you're actually good to go as long as the HR group is satisfied with the spec, rather than because you did a review, and the HR group may well continue tracking changes to documents for several years after the review. ;-)

>    my current thinking is to stick to the current wording in the proposed charter, without closing this issue. If new wording emerges before the end of the AC review, we can always update the proposed charter imho.

> That sounds like the wrong thing to do. What r12a wants is not what is in the charter. I (and fantasai) don't like what's in the charter. r12a doesn't like what fantasai proposed, but he's not defending the charter either. Until we can agree on something (and I am hopeful that we can, since r12a's points look reasonable to me), we should not place on ourselves constraints beyond what the Process asks. I suggest deleting that text until new wording emerges.

So here's another attempt, starting from Fantasai's text:

> Invitation for review must be issued in conjunction with FPWD, should be issued when major changes occur in a specification, and must be issued 3 months before a planned CR transition date. As soon as horizontal groups are satisfied that the spec meets their needs, the CR transition date can be brought forward, if appropriate.

**Comment by fantasai** [comment]
@r12a CSSWG doesn't plan a CR transition date until it has a spec that is "substantially the same" as the one it takes to CR, so I don't understand how your suggestion is any different from mine except insofar that it requires the request for review to be issued on a very particular date.

**Comment by r12a** [comment]
If CSS is unable/unwilling to plan 3 months in advance, then yes i suppose this wording will not make much difference for CSS specs, and you'll need to accept that your schedules are going to be extended to allow for horizontal review, and perhaps in some cases you'll have more rework to do than if you had involved horizontal groups before finalising the spec.  (That said, i18n at least does tend to get involved in many CSS specs earlier than others.) But I don't see this text as CSS specific. Most of the working groups we deal with do seem to have specific dates in mind. They, by their own admission, just don't think of horizontal review until the very last minute, when they tend to have a hard stop staring them in the face.

In addition, by setting a less vague timeframe for the review request to be issued, the edits i made will allow automated tools to notify the horizontal groups of the need to collaborate well before the typical 2-3 weeks we are used to.   Such tools could also be made to prevent a CR date being set that is less than 3 months away (although, as the text says, it may be possible to reduce that 3 month window if all goes well).

**Comment by fantasai** [comment]
@r12a

> If CSS is unable/unwilling to plan 3 months in advance, then yes i suppose this wording will not make much difference for CSS specs

Yes, it does. It means that if we decide a spec is ready for CR, regardless of whether it's received HR or not, if we didn't happen to send a request for review within the last 2.5 months or so, we have to send another request for review and schedule a CR date exactly 3 months ahead of that request.

> you'll need to accept that your schedules are going to be extended to allow for horizontal review

We don't have a schedule, therefore it can't be extended.

> perhaps in some cases you'll have more rework to do than if you had involved horizontal groups before finalising the spec

Perhaps we should have sent requests for review before deciding we wanted to go to CR? Isn't that how it's *supposed* to work?

> Most of the working groups we deal with do seem to have specific dates in mind. They, by their own admission, just don't think of horizontal review until the very last minute, when they tend to have a hard stop staring them in the face.

I don't see why you want to make this our problem.

> the edits i made will allow automated tools to notify the horizontal groups of the need to collaborate well before the typical 2-3 weeks we are used to

What exactly are you keying these automated tools off of? As I said, CSS doesn't have a CR publication schedule a human can read, nevermind an automated tool. The only point at which we set a CR publication date is after we've resolved to go to CR: we fill one into the publication request.

**Comment by frivoal** [comment]
@plh To be clear: despite the label you placed on this issue, I don't count this as an enhancement. I want this issue addressed before the charter is set for AC review.

Ideally, we will find wording that is agreeable to @r12a, @fantasai, and myself, (and anyone else interested in the matter). If we don't, and until we do, the current wording should be deleted from the charter. It does not accomplishes what @r12a is asking for on behalf of horizontal groups, yet constrains the working group beyond what requirements of the process.

**Comment by frivoal** [comment]
@r12a 
> An easy way to fix that is to remove 'at least' from 'at least 3 months' , and it could be made clearer by saying 3 months before the expected CR transition date – something i was inclined to suggest anyway, actually

I would very strongly (i.e. formally) object to that. This would mean delaying all CR transitions by 3 months, which is not acceptable:
* If we know how to fix all issues, we just do so, and so we are ready for CR. Not in 2 or 7 months. Now.
* If we don't know how to fix all issues, then we don't know.
The only point when we know CR is going to happen is when we're ready for it. We may hope to be ready soon, we may try harder for some high priority specs than some others. But we don't never have a date that is known in advance. Adding 3 months every time, regardless of how recently and how thoroughly the spec has already received horizontal, is not acceptable. Also, it guarantees that Horizontal groups would never get more than 3 months, because at that point, we'd always be waiting for you, and would not be inclined to give longer delays, nor would we be in a very receptive mindset, because at that point, we'd have decided that we were done (since we're ready to call for CR).

The goal, as you pointed out in an earlier comment, should be to ensure that Working Groups and Horizontal groups collaborate on an ongoing basis, and do good work together as much as possible, avoiding last minute rushes. Not to make horizontal review groups become a blocking source of delays.

I would encourage you to reconsider the wording proposed by @fantasai in https://github.com/w3c/charter-drafts/issues/219#issuecomment-521433236, it seems to me that it supports the collaboration mode you advocate for quite well:

> Invitation for review must be issued in conjunction with FPWD, should be issued when major changes occur in a specification, and must be issued with sufficient time (which should be at least three months, unless the horizontal review group has indicated otherwise) before a CR transition date that a review can be completed on substantially the same document that will become the CR.

* "Should ask for review when major changes occur" is conductive on ongoing collaboration.
* "at least 3 months before CR, unless the Horizontal Groups have said otherwise" does not mean we need to systematically ask the horizontal groups. It means they get 3 months, unless they waive it. Which I expect they will do either when they have completed the review in less than 3 months, or if they have been in a sufficiently collaborative mode that no extra review is needed.
* "on substantially the same document that will become the CR". This means that the WG can't just claim horizontal review has happened if the draft has been completely overhauled since last review. But it does say substantially, rather than exactly, so that WG does not have to wait until everything is finished before asking, which would bring us back the Last Call way of doing things instead of ongoing collaboration.

**Comment by frivoal** [comment]
I note that despite lack of consensus on the current wording, the charter has been proposed to AC voting. None of the participants in the discussion about it support the current wording, but it is included in the proposed charter. I suppose I need to file a formal objection now :(

**Comment by r12a** [comment]
> I don't see why you want to make this our problem.

I've been trying to find some wording that will go in the template rather than just in the CSS charter. But i don't want to block the CSS charter, so perhaps i should return to the generic version later.

> This would mean delaying all CR transitions by 3 months, which is not acceptable:

That's not what my text says; see the 2nd sentence.

> "at least 3 months before CR, unless the Horizontal Groups have said otherwise" does not mean we need to systematically ask the horizontal groups. It means they get 3 months, unless they waive it. Which I expect they will do either when they have completed the review in less than 3 months, or if they have been in a sufficiently collaborative mode that no extra review is needed.

I suspect we're trying to say the same thing, but interpreting each other's text differently.

Here is Fantasai's original suggestion:

> Invitation for review must be issued in conjunction with FPWD, should be issued when major changes occur in a specification, and must be issued with sufficient time (which should be at least three months, unless the horizontal review group has indicated otherwise) before a CR transition date that a review can be completed on substantially the same document that will become the CR.

1. 'unless the horizontal review group has indicated otherwise' still sounds ambiguous to me. I'd be happy to say 'unless the horizontal review group requires less time'.  Otherwise, please suggest an alternative wording.
2. 'that a review can be completed on substantially the same document that will become the CR'. I still think (as do other HR folks i spoke with) that this implies leaving the review too late. I suspect we have different interpretations of the meaning of 'substantially the same', and that others will too.  I propose to just drop that text. Do you have a particular reason to keep it?

**Comment by frivoal** [comment]
> I suspect we're trying to say the same thing, but interpreting each other's text differently.

Me too. I am not under the impression that we have conflicting goals. We just need to find text that we all think actually achieves these goals without unpleasant side effects.

> > This would mean delaying all CR transitions by 3 months, which is not acceptable:
> That's not what my text says; see the 2nd sentence.

That's not what your text says, but it seems likely to be a consequence of it, despite the second sentence:
* There is generally no planned CR transition date and no ability to pick a date until the document is actually ready for a CR transition. In some cases there might be, but in the general case there is not. We generally do get a sense that we're getting close, but that's different from having a planned transition date. So with "exactly 3 months before the date" requirements, we cannot call for horizontal review until after everything is finished. I'm pretty sure that's not what you want, but I don't know how to satisfy the text otherwise. With "at least 3 months", that problem goes away, but then we're at risk of a 4 year old review of a document that has radically changed since being considered OK.
* The second sentence means that this 3 month period may occasionally be shortened when HR groups have been able to be fast. That's good, but doesn't change the fact that we're tacking a mutli-month period at the end, which is not good, both because it's at the end (which you rightfully don't like), and because it's months of waiting.

> unless the horizontal review group has indicated otherwise' still sounds ambiguous to me. I'd be happy to say 'unless the horizontal review group requires less time'. Otherwise, please suggest an alternative wording.

To me, what this tries to say is that if the HR group comes back with "we only needed 2 months",  or "actually, we're good, we just looked at it since we have been continuously in contact", then we don't need to wait 3 months. But if they don't, then 3 months is the default. Your sentence here seems to say the same thing, but I don't think it is clearer. I willing to try another rephrasing, but first, you said you though fantasai's phrasing was ambiguous. Could you clarify what (undersirable) meaning you think can be read into it?


> 'that a review can be completed on substantially the same document that will become the CR'. I still think (as do other HR folks i spoke with) that this implies leaving the review too late. I suspect we have different interpretations of the meaning of 'substantially the same', and that others will too. I propose to just drop that text. Do you have a particular reason to keep it?

I think this tries to prevent excesses in either directions:
* make sure we don't implicitly require the review to happen on **exactly the same** document, which would indeed be too late. In the absence of any qualification, it can be reasonable to claim that unless the review was on the actual final document, then the actual final document hasn't been reviewed, and so we need another one. By saying "substantially the same", we allow for some latitude, and can do it when we're close to wrapping up, but before everything is nailed down.
* Clarify that if the document was reviewed punctually at some point in the past but was radically modified thereafter, that's not good enough.

What fantasai's wording doesn't do is call for continuous cooperation, which you say is desirable (and I agree). However, it does not prevent it either, and allows the HR group to waive or reduce their final 3 month period if they are satisfied that they've reviewed the document sufficiently. I'll also note that the wording you've proposed (or the one currently in the charter) do not call for this continuous cooperation either.

I may tweak further based on your answers to the points above, but how about:

> Invitation to collaborate on and review the specification must be issued in conjunction with FPWD. To facilitate this collaboration, further notices should be issued when major changes occur in a specification, and must be issued with sufficient time (which should be at least three months, unless the horizontal review groups have indicated they needed less) before a CR transition that horizontal review groups can work with the Working Group to ensure that the document that will become the CR has received sufficient review.

**Comment by therealglazou** [comment]
Copy of a message I sent earlier today to w3c-css-wg:

> Hi there,
> 
> I would like to share with the Group - and Team - my concerns about section 4 of the proposed new Charter. In short:
> 
> - WG should seek horizontal review from SIX different groups at each and every publication step of each spec
> 
> - for non-CR, call for review should happen at transition time
> 
> - for CR, call for review should happen at least 3 months before transition/publication
> 
> With an ex-CSS WG Co-chair's eye, I see this section as badly shaped prose in answer to a perfectly valid request:
> 
> - in the past, getting even one single horizontal review on time has been extremely complicated
> 
> - getting 6 different reviews on time assumes there are enough resources at the right moment in each of these groups, at any time of the year, to perform reviews in a timely manner. Let's not lie to ourselves, this is not the case.
> 
> - asking reviews 3 months before CR will imply that CR totally stalls 3 months since any change would trigger another set of reviews. I don't think this is going to work. On another hand, I'm sure it will generate a lot of frustration given how are incubated (and early implemented and shipped) features these days.
> 
> - since the Process section 6.4 requests "wide review" including horizontal reviews to advance a document to CR, a late horizontal review would then block CR release
> 
> - how could we even launch the 3 months timer for CR release? The only way to do that would be to move a last WD having all positive horizontal reviews to CR without any change, reintroducing de facto the LCWD status. But that also supposes the reviews did happen...
> 
> - why does that belong to the CSS WG Charter anyway? This is Process-level prose that is not specific to CSS.
> 
> - this adds time and complexity to our process. The main complaint about W3C is "W3C is slow and painful". Under that light, adding 3 months of uncertainty to the loop is really weird.
> 
> The whole thing seems to me weak and adds too many uncertainties to the spec release process. It does not feel right and I think it should be revamped or, even better, removed.
> 
> </Daniel>

**Comment by brewerj** [comment]
Several thoughts: 

1. This thread seems mis-named. My understanding of the horizontal review teams' intention is to get _clear advance notice_ so that we can do timely reviews, not to delay anybody's spec by 3 months. The issue title, IMO, is the opposite our of intention with regard to improving the HR process. How about re-titling this to "advance notice for horizontal reviews." 
2. In exploring different ways to meet the needs of a likely upcoming Ever* process -- and to improve W3C's review processes generally -- we have been looking at ways to make the tooling better support more granular notification of individual feature changes and additions. This could remove the need for horizontal review teams to re-review entire specs multiple times; they could review most of a spec initially, then just the individual changes as the spec progresses, making the review process more agile. Yes, we'd need to push to get new tooling in place sooner than planned, but it could be valuable for more timely horizontal reviews, and perhaps also for other aspects of more agile W3C spec-development.
3. +1 to Glazou's point that this should be handled as a Process-level discussion.

**Comment by frivoal** [comment]
Thanks for the feedback @brewerj 

> This thread seems mis-named

By "mandatory delays for horizontal Review", I meant "the charter imposes mandatory delays for horizontal Review and that's a problem", not "we should impose mandatory delays for horizontal Review". I'm ok with trying to frame the discussion in a more positive way, but I don't want to lose track of the fact that, despite good intentions, the wording currently in the charter is problematic, and that's an issue.

> [... your point 2 ...]

I agree. I've attempted to move away from this notion necessarily having multiple reviews, and instead saying that the document needs review in general, but without characterizing that as multiple punctual acts in the text I proposed at the bottom of https://github.com/w3c/charter-drafts/issues/219#issuecomment-525565381

> +1 to Glazou's point that this should be handled as a Process-level discussion.

I agree as well. Would you be OK with deleting this specific wording from this specific charter and moving this issue to the Process CG to solve it centrally?

**Comment by samuelweiler** [comment]
Hard deadlines don't belong in the charter.  With the caveat that I think most charters are too verbose and we should be putting stuff like this in the Process, not in charters: I want the charter to (at most) encourage the behavior we want to see (namely, collaboration at useful times).   I've created a PR with some suggested text.  I also want to see failure to get review be a blocking issue, but that's a process thing.

**Comment by frivoal** [comment]
I fully support the PR proposed by @samuelweiler , and continuing the rest of this discussion in the ProcessCG to see what can/should be added to the Process. Speaking of the process, this may deserve its own issue, but a good place to start is probably by reviewing and contributing the existing https://github.com/w3c/w3process/issues/130, which is either the same topic, or at least overlaps a lot.

**Comment by plehegar** [comment]
@r12a 12a @michael-n-cooper , your input would be appreciated

**Comment by michael-n-cooper** [comment]
I think Sam's proposal strikes a good balance. I agree with the need for sufficient timeline to conduct reviews, need to review at key stages (more clearly spelled out in the proposal), and the need for review groups to know about changes to avoid re-reviewing from scratch each time. I also understand that it's a bit much to make mandatory timelines related to all this, and that the guidance should be "best practice". What matters is obtaining the review, and if it can be done quicker or in a different way than the best practice, there shouldn't be a constraint on advancement.

I have more complicated thoughts on whether this sort of guidance belongs in charters or in a support resource. The guidance reads like a best practice that would be best in a horizontal review documentation, with components of it even in the Process doc to set out uniform requirements. As a person who writes charters, I'm reluctant to add large amounts of boilerplate text because I worry that they are easily recognized as boilerplate and ignored. That said, we have found that having an overly generic and simple horizontal review boilerplate that references external text is also too easily ignored, which is what to adding more explicit requirements to charter boilerplates. As a person who implements charters, I do find that I look at the charter at key transition points to make sure I haven't missed anything, so having explicit guidance right there is helpful.

**Comment by r12a** [comment]
Fine by me.

**Comment by plehegar** [comment]
This issue has been resolved. See latest wording
[[
For all specifications, this Working Group will seek horizontal review for accessibility, internationalization, performance, privacy, and security with the relevant Working and Interest Groups, and with the TAG. Invitation for review must be issued during each major standards-track document transition, including FPWD. The Working Group is encouraged to engage collaboratively with the horizontal review groups throughout development of each specification. The Working Group is advised to seek a review at least 3 months before CR and is encouraged to proactively notify the horizontal review groups when major changes occur in a specification following a review.
]]

Re-open this issue or open a new one if you feel differently.

**Comment by nigelmegitt** [comment]
This still has the recommendation to leave 3 months before every CR transition, which seems unhelpful. Address this as:

s/at least 3 months before CR/at least 3 months before first entering CR

**Comment by nigelmegitt** [comment]
(I don't seem to have rights on this repo to reopen issues)

**Comment by plehegar** [comment]
the charter was approved in early October. See also #240

**Comment by svgeesus** [comment]
re-opening with edited title and tagged "template"

**Comment by svgeesus** [comment]
I checked, and the edits have all been made to the template.
